File: Finding Subset.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
	 long long int n;
	 cin>>n;
	 vector <long long int >v;
	 list<long long int >l;
	 v.reserve(n);
	 for(long long int i=0;i<n;i++)
	 {
	 	long long int d;
	 	cin>>d;
	 	v.push_back(d);
	 	l.push_back(d);
	 }
	 sort(v.begin(),v.end());
	 vector<long long int>::iterator it; 
  	 it = v.begin(); 
	 if(v[0]==1)
	 {
	 	for(long long int i=1;i<v.size();i++)
	 	{
	 		double mm=cbrt(v[i]);
	 		if(mm==int(mm))
	 		{
	 			l.remove(v[i]);
	 			v.erase(it+i);
	 			
			 }
		}
	 }
	 reverse(v.begin(),v.end());
	 long long int cc=0;
	 for(long long int i=0;i<v.size();i++)
	 {
	 	int flag=1;
	 	for(long long int j=0;j<v.size();j++)
	 	{
	 		if(i!=j and v[i]!=v[j])
	 		{
	 			double tmp=cbrt(v[i]*v[j]);
	 			
	 			if(tmp==int(tmp))
	 			{
	 				long long int mm=min(v[i],v[j]);
	 				list <long long int>:: iterator it;
					it=find(l.begin(),l.end(),mm);
	 				if(it!=l.end())
	 				{
	 					l.remove(mm);
					 }
				}
			}
	 		
		}
	 }
	 cout<<l.size();
}

File: Finding Subset.py
from itertools import combinations
import math
n=int(input())
ls=[]
while(n>0):
    t=int(input())
    ls.append(t)
    n-=1
ls.sort()
tmp1=[]
if(ls[0]==1):
    tmp1.append(1)
    for i in range(1,len(ls)):
        if(int(round(ls[i] ** (1. / 3))) ** 3 == ls[i]):
            continue
        else:
            if ls[i] not in tmp1:
                tmp1.append(ls[i])
res=[]
tmp1.sort(reverse=True)
comb=list(combinations(tmp1,2))
rem=[]
for item in comb:
    tmp=list(item)
    x=tmp[0]*tmp[1]
    if(int(round(x ** (1. / 3))) ** 3 == x):
        rem.append(min(tmp[0],tmp[1]))
    else:
        if tmp[0] not in res:
            res.append(tmp[0])
        if tmp[1] not in res:
            res.append(tmp[1])
for item in rem:
    if item in res:
        res.remove(item)
cc=0
for item in res:
    cc+=ls.count(item)
print(cc)
